astropy/cosmology/tests/test_cosmology.py

```diff
@@ -1582,6 +1582,46 @@ def test_z_at_value():
             z_at_value(cosmo.angular_diameter_distance, 1500*u.Mpc, zmin=4.)
 
 
+@pytest.mark.skipif('not HAS_SCIPY')
+@pytest.mark.parametrize('method', ['Brent', 'Golden', 'Bounded'])
+def test_z_at_value_bracketed(method):
+    """
+    Test 2 solutions for angular diameter distance by not constraining zmin, zmax,
+    but setting `bracket` on the appropriate side of the turning point z.
+    Setting zmin / zmax should override `bracket`.
+    """
+    z_at_value = funcs.z_at_value
+    cosmo = core.Planck13
+
+    if method == 'Bounded':
+        with pytest.warns(UserWarning, match=r'fval is not bracketed'):
+            z = z_at_value(cosmo.angular_diameter_distance, 1500*u.Mpc, method=method)
+        if z > 1.6:
+            z = 3.7914908
+            bracket = (0.9, 1.5)
+        else:
+            z = 0.6812777
+            bracket = (1.6, 2.0)
+        with pytest.warns(UserWarning, match=r"Option 'bracket' is ignored"):
+            assert allclose(z_at_value(cosmo.angular_diameter_distance, 1500*u.Mpc, method=method,
+                                       bracket=bracket), z, rtol=1e-6)
+    else:
+        with pytest.warns(UserWarning, match=r'fval is not bracketed'):
+            assert allclose(z_at_value(cosmo.angular_diameter_distance, 1500*u.Mpc, method=method,
+                                       bracket=(0.9, 1.5)), 0.68127769625288614, rtol=1e-6)
```

Comment:
with the rtol relatively low... don't need so many decimal places?

Reply:
Not really, just copied from the old tests.

Reply:
doesn't need fixing anyway.

Reply:
But could make it a bit easier to relate to the backwards test for 'Bounded'...